### PR TITLE
Make adding tabs to DetailsPanel easier

### LINKED ISSUE #81: Refactor tab switching code
Refactor code to make adding new tabs easier

### PR DESCRIPTION
**Pull request by chesterhow**
### Adding a new tab
Adding a new tab to DetailsPanel can now be done as follows:
1. Create a new `fxml` file for the new tab UI and its corresponding `java` file
2. Add an enum for the new tab to `DetailsPanelTab.java`
3. Pass relevant information to `DetailsPanel` on initialisation via `MainWindow` if necessary
4. Update `DetailsPanel.java#toggleTab()`, by adding your new tab to the switch case

Fixes #81

### CODE REVIEW

File: src/main/java/seedu/address/ui/DetailsPanel.java [hunk 1]
```diff
@@ -1,62 +1,33 @@
...
 
     @FXML
-    private ListView<Person> personDetailsListView;
+    private VBox tabPlaceholder;
 
     /**
      * Creates a {@code DetailsPanel} with the given {@code ObservableList}s.
-     * @param upcomingDatesList A list of upcoming dates.
+     * @param upcomingEventsList A list of upcoming events.
      * @param detailedPerson A list containing a single {@code Person}.
```
Comment (ivantjh) [inline src/main/java/seedu/address/ui/DetailsPanel.java:24]: I think would be good to document something the rationale for this

**Review by ivantjh — APPROVED** [review]
lgtm!

File: src/main/java/seedu/address/ui/DetailsPanel.java (hunk 1 above)
Comment (chesterhow) [reply src/main/java/seedu/address/ui/DetailsPanel.java:24]: I realised in some places I named it `UpcomingDates` while in others they were `UpcomingEvents`. Actually, I just realised in `logic` it's still called `UpcomingDates`. I think I'll just revert and switch it all to `UpcomingDates`.

Comment (ivantjh) [reply src/main/java/seedu/address/ui/DetailsPanel.java:24]: Ok, btw I was talking about the usage of a single `Person` in the `detailedPerson` observable list. That one probably need to document the rationale for it.

Comment (chesterhow) [reply src/main/java/seedu/address/ui/DetailsPanel.java:24]: Ok will do. Anyway I think i'll leave it as `UpcomingEvents` for now. Might rename all in another PR

**Pull request merged**